float32 startX
float32 startY
float32 endX
float32 endY

float32 gapWidth

#startX- x coordinate of starting point of gap
#startY - y coordinate of starting point of gap
#endX- x coordinate of endpoint of gap
#endY - y coordinate of endpoint of gap
#gapWidth - width of gap in terms of euclidean distance between start and end points of gap
